custom/icds_reports/reports/issnip_monthly_register.py

```diff
@@ -287,4 +288,32 @@ def agg_child_health_monthly(self):
             }, 'rations_21_plus_distributed')),
 
         )
-        return data[0] if data else None
+        return data
+
+    def get_awc_name(self, awc_id):
+        return SQLLocation.objects.get(location_id=awc_id).name,
+
+    @cached_property
+    def to_pdf_format(self):
+        for awc in self.config['awc_id']:
+            agg_awc_monthly_data = [x for x in self.agg_awc_monthly_data if x['awc_id'] == awc]
```

Comment:
This would be much nicer/faster if you created a dictionary `{awc_id: data}` for each of these